src/coreclr/src/jit/gentree.h

```diff
@@ -4526,17 +4663,20 @@ struct GenTreeJitIntrinsic : public GenTreeOp
 /* gtSIMD   -- SIMD intrinsic   (possibly-binary op [NULL op2 is allowed] with additional fields) */
 struct GenTreeSIMD : public GenTreeJitIntrinsic
 {
-    SIMDIntrinsicID gtSIMDIntrinsicID; // operation Id
 
     GenTreeSIMD(var_types type, GenTree* op1, SIMDIntrinsicID simdIntrinsicID, var_types baseType, unsigned size)
-        : GenTreeJitIntrinsic(GT_SIMD, type, op1, nullptr, baseType, size), gtSIMDIntrinsicID(simdIntrinsicID)
+        : GenTreeJitIntrinsic(GT_SIMD, type, op1, nullptr, baseType, size)
     {
+        gtSIMDIntrinsicID = simdIntrinsicID;
+        assert(gtSIMDIntrinsicID == simdIntrinsicID);
```

Comment:
what does this assert check?

Reply:
It checks that the intrinsic ID fits into the the allocated space. At one time I'd made this smaller, before I restructured a bit. This could now be deleted.
Note, though, that `GenTreeHWIntrinsic` and `GenTreeSIMD` which derive from `GenTreeJitIntrinsic` are small nodes. I think we just allocate a large node for GT_INTRINSIC because it might become a call.